great_expectations/experimental/datasources/spark_google_cloud_storage_datasource.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
+
+import pydantic
+from typing_extensions import Literal
+
+from great_expectations.core.util import GCSUrl
+from great_expectations.experimental.datasources import _SparkFilePathDatasource
+from great_expectations.experimental.datasources.data_asset.data_connector import (
+    DataConnector,
+    GoogleCloudStorageDataConnector,
+)
+from great_expectations.experimental.datasources.interfaces import TestConnectionError
+from great_expectations.experimental.datasources.spark_datasource import (
+    SparkDatasourceError,
+)
+from great_expectations.experimental.datasources.spark_file_path_datasource import (
+    CSVAsset,
+)
+
+if TYPE_CHECKING:
+    from google.cloud.storage.client import Client as GoogleCloudStorageClient
+    from google.oauth2.service_account import (
+        Credentials as GoogleServiceAccountCredentials,
+    )
+
+    from great_expectations.experimental.datasources.interfaces import (
+        BatchSorter,
+        BatchSortersDefinition,
+    )
+
+
+logger = logging.getLogger(__name__)
+
+
+GCS_IMPORTED = False
+try:
+    from google.cloud import storage
+    from google.oauth2 import service_account
+
+    GCS_IMPORTED = True
+except ImportError:
+    storage = None
+    service_account = None
```

Comment:
Why do you need to set these to `None` at all?

Reply:
@Kilo59 Otherwise, we get a `PyCharm` waning, "service_account in the try block with except ImportError should also be defined in the except block."  What should we do about all these situations?  In the past, we have handled them using "variable = None" in the except block.  Should we simply disable those `PyCharm` warnings?  Thanks.

Reply:
If the only reason is "because PyCharm" complained then yes, I would get rid of them.
You can know if these values are available by checking `GCS_IMPORTED`.
And if we only need them for type-checking they should be part of the `if TYPE_CHECKING` block.

Reply:
@Kilo59 I am looking for a way to disable these warnings in `PyCharm` -- thanks.

Reply:
@Kilo59 I found a way to disable these warnings: `  # noqa: disable=E0602`.  Thanks.